src/main/java/com/opentok/AudioStreamerConnectionProperties.java

```diff
@@ -0,0 +1,188 @@
+	 *
+	 * @return "websocket".
+	 */
+	public String type() {
+		return "websocket";
+	}
+
+	protected AudioStreamerConnectionProperties(Builder builder) {
+		this.uri = Objects.requireNonNull(builder.uri);
+		this.streams = builder.streams.isEmpty() ? null : Collections.unmodifiableCollection(builder.streams);
+		this.headers = builder.headers.isEmpty() ? null : Collections.unmodifiableMap(builder.headers);
+	}
+
+	/**
+	 * Intermediary stateful object used to construct {@link AudioStreamerConnectionProperties}.
+	 */
+	public static class Builder {
+		private URI uri;
+		private Collection<String> streams = new ArrayList<>();
+		private Map<String, String> headers = new HashMap<>();
+
+		/**
+		 * Call this method and pass in the uri where your service is listening.
+		 * This parameter is mandatory.
+		 *
+		 * @param uri A valid URI.
+		 *
+		 * @return The ConnectProperties.Builder object with the uri setting.
+		 */
+		public Builder uri(URI uri) {
+			this.uri = uri;
+			return this;
+		}
+
+		/**
+		 * Call this method and pass in the uri where your service is listening.
+		 *
+		 * @param uri A valid string representation of a URI.
+		 *
+		 * @return The ConnectProperties.Builder object with the uri setting.
+		 */
+		public Builder uri(String uri) {
+			return uri(URI.create(uri));
+		}
+
+		/**
+		 * Sets the collection of stream IDs to the parameter.
+		 *
+		 * @param streams The collection of stream IDs.
+		 *
+		 * @return This builder with the streams property set.
+		 */
+		public Builder streams(Collection<String> streams) {
+			this.streams = Objects.requireNonNull(streams);
+			return this;
+		}
+
+		/**
+		 * Adds the stream ID to the streams property.
+		 *
+		 * @param stream The Stream ID to add.
+		 *
+		 * @return This builder with the additional stream ID.
+		 */
+		public Builder addStream(String stream) {
+			if (StringUtils.isBlank(stream)) {
+				throw new IllegalArgumentException("Stream ID cannot be blank");
+			}
+			streams.add(stream);
+			return this;
+		}
+
+		/**
+		 * Adds the stream IDs to the streams property.
+		 *
+		 * @param streams The stream IDs to add.
+		 *
+		 * @return This builder with the additional stream IDs.
+		 */
+		public Builder addStreams(String... streams) {
+			return addStreams(Arrays.asList(streams));
+		}
+
+		/**
+		 * Adds the stream IDs to the streams property.
+		 *
+		 * @param streams The collection of stream IDs to add.
+		 *
+		 * @return This builder with the additional stream IDs.
+		 */
+		public Builder addStreams(Collection<String> streams) {
+			this.streams.addAll(Objects.requireNonNull(streams));
+			return this;
+		}
+
+		/**
+		 * Sets the map of additional headers to the parameter.
+		 *
+		 * @param headers The map of header properties.
+		 *
+		 * @return This builder with the headers property set.
+		 */
+		public Builder headers(Map<String, String> headers) {
+			this.headers = Objects.requireNonNull(headers);
+			return this;
+		}
```

Comment:
Also, is there an advantage to having both `headers(Map<String, String> headers)`, and addHeader(String: key, String: value)? (OK if there is.)